packages/mask/src/plugins/RedPacket/SNSAdaptor/hooks/useCreateCallback.tsx

```diff
@@ -111,17 +115,21 @@ export function useCreateParams(redPacketSettings: RedPacketSettings | undefined
     return getCreateParams
 }
 
+export function useCreateParams(redPacketSettings: RedPacketSettings, version: number, publicKey: string) {
+    const getCreateParams = useCreateParamsCallback(redPacketSettings, version, publicKey)
+    return useAsync(() => getCreateParams(), [redPacketSettings, version, publicKey])
+}
+
 export function useCreateCallback(redPacketSettings: RedPacketSettings, version: number, publicKey: string) {
     const account = useAccount(NetworkPluginID.PLUGIN_EVM)
     const chainId = useChainId(NetworkPluginID.PLUGIN_EVM)
     const redPacketContract = useRedPacketContract(chainId, version)
-    const getCreateParams = useCreateParams(redPacketSettings, version, publicKey)
+    const { value: createParams } = useCreateParams(redPacketSettings, version, publicKey)
     const connection = useWeb3Connection(NetworkPluginID.PLUGIN_EVM)
     const web3 = useWeb3(NetworkPluginID.PLUGIN_EVM)
 
     return useAsyncFn(async () => {
         const { token } = redPacketSettings
-        const createParams = await getCreateParams()
```

Comment:
Why do this change?
Would not allow to  create if `createParams` is not ready, that makes the button look unresponsive.

Reply:
https://github.com/DimensionDev/Maskbook/pull/6467/files/6b1f702400fbd1a91f06b425ae34f0d4a382713d#diff-bfc6b6815b9320fef224127c98b0e33f066453ad89d960ea514385b0e821a5daR127